src/Storage/Notmuch.hs

```diff
@@ -31,9 +35,50 @@ getMessages s settings = do
         mails <- mapM messageToMail msgs
         return $ Vec.fromList mails
 
+setNotmuchMailTags
+    :: FilePath
+    -> NotmuchMail
+    -> IO (Either String NotmuchMail)
+setNotmuchMailTags dbpath m =
+    case mailTagsToNotmuchTags m of
+        Nothing -> pure $ Left ("Tags are corrupt")
+        Just nmtags ->
+            bracket
+                (databaseOpen dbpath >>= either (error . show) pure)
```

Comment:
I've fiddled with the `error` here. This would obviously crash, but the Type expected in IO is always (Database RW), but I'm puzzled, since I thought the first function handles the Left value, which is `String`. What I want is to give an error back, rather than... well just crashing.

Reply:
We're going to need to start using MonadError monad transformer to be able to better manage *IO-with-error* scenarios.  For now leave it as-is and I'll address in a later commit (just add a FIXME comment)

Reply:
I've moved it in a backlog item... when we have the time.